File: StockExchange/StockExchange.Server/DTOs/StockDto.cs
﻿using StockExchange.Server.Model;

namespace StockExchange.Server.DTOs
{
    public class StockDto
    {
        public string Symbol { get; set; }
        public double Price { get; set; }
        public DateTime Timestamp { get; set; }
        public int Quantity { get; set; }

        public StockDto()
        {

        }

        public StockDto(Stock stock)
        {
            this.Symbol = stock.Symbol;
            this.Price = stock.Price;
            this.Timestamp = stock.Timestamp;
            this.Quantity = stock.Quantity;
        }
    }
}

File: StockExchange/stockexchange.client/src/app/StockDto.ts
export class StockDto
{
  public price: number = 0;
  public symbol: string = "";
  public timestamp: string = "";

  constructor(price: number, symbol: string, timestamp: string) {
    this.price = price;
    this.symbol = symbol;
    this.timestamp = timestamp;
  }

}
